bread/layout/components/datatable.py

```diff
@@ -133,20 +135,30 @@ def __init__(
                 hg.TH(headcontent, **getattr(col.header, "td_attributes", {}))
             )
 
-        self.iterator = hg.Iterator(
-            row_iterator,
-            rowvariable,
-            hg.TR(
-                *[
-                    hg.TD(col.cell, **getattr(col.cell, "td_attributes", {}))
-                    for col in columns
-                ]
-            ),
+        row_template = hg.TR(
+            *[
+                hg.TD(col.cell, **getattr(col.cell, "td_attributes", {}))
+                for col in columns
+            ]
         )
+
+        # combining Formsets with Datatables needs to handle iterations differently,
+        # the formsetfield needs to control the container in order to attach new entries/rows
```

Comment:
that's a very useful comment, thanks!
(hopefully not needed anymore if we can have a cleaner design as I suggested in other comments)